### PR TITLE
Update doxygen start page

### PR DESCRIPTION
**Pull request by mayrmt**
## Description and Context

Replace outdated content on the doxygen starting page with a minimal introduction to 4C as well as some reference to other useful resources, in particular our user guide.

## Related Issues and Merge Requests

- Closes https://gitlab.lrz.de/baci/baci/-/issues/1285

### CODE REVIEW

**Review by c-p-schmidt — APPROVED** [review]

File: doc/doxygen/index.doc [hunk 1]
```diff
@@ -1,265 +1,20 @@
...
-\section drt_intro Introduction
+\section Overview
 
-DRT provides fundamental services to construct and manage 2D/3D discretizations
-in parallel and serial. LinAlg provides parallel sparse and dense matrix/vector operations.
+4C (Comprehensive Computational Community Code) is a powerful research code for multi-physics computer simulations.
+Building on more than 20 years of research and development, 4C can model a plethora of physical systems, including solid mechanics, fluid dynamics, scalar transport, and chemical reaction among others.
 
-\section Overviewdrt Overview of DRT
+Large parts of 4C are based on finite element methods (FEM), but alternative discretization methods such as discontinuous Galerkin methods (DG), particle methods and mesh-free methods have also been successfully integrated. The research software is implemented throughout in object-oriented programming (C++) using modern software design and is parallelized with MPI for distributed memory hardware architectures.
```
Comment (amgebauer) [inline doc/doxygen/index.doc:10]:
```suggestion
Large parts of 4C are based on finite element methods (FEM), but alternative discretization methods such as discontinuous Galerkin methods (DG), particle methods and mesh-free methods have also been successfully integrated. The research software is implemented throughout in C++ using modern software design and is parallelized with MPI for distributed memory hardware architectures.
```
We are using a lot object-oriented programming but that's in many cases not the best design decision.

File: doc/doxygen/index.doc [hunk 2]
```diff
@@ -1,265 +1,20 @@
...
-\author Michael W. Gee (mwgee@sandia.gov), Sandia National Laboratories, Org. 1414.
-
-I would like to thank Barbara Wohlmuth, Stefan Huuber, Raymond Tuminaro and
-Pavel Bochev for discussions and support in creating this work.
-
-<br>
-<br>
-*/
--->
+- Source Code Repository on GitHub: https://github.com/4C-multiphysics/4C
```
Comment (amgebauer) [inline doc/doxygen/index.doc:20]:
```suggestion
- Website: https://www.4c-multiphysics.org
- Source Code Repository on GitHub: https://github.com/4C-multiphysics/4C
- Readthedocs: https://4c-multiphysics.github.io/4C/readthedocs/
```

**Review by amgebauer — APPROVED** [review]
Thanks for cleaning up the Doxygen main page.

File: doc/doxygen/index.doc (hunk 1 above)
Comment (mayrmt) [reply doc/doxygen/index.doc:10]: @amgebauer Sure, makes sense! But do we really use "modern software design"? Maybe not... what do you think?

Comment (amgebauer) [reply doc/doxygen/index.doc:10]: True, for me it's just a (often overused) phrase. What is modern now is typically outdated in a few years. I think we have things that one can call "modern", but we also have a lot of things that has been called "modern" more than a decade ago. Still, I don't see a direct need to actively remove it. It's just the doxygen start page, and we, for sure, aim at modern software design.

**Pull request merged**